src/components/views/elements/PowerSelector.js

```diff
@@ -43,15 +40,23 @@ module.exports = React.createClass({
 
     getInitialState: function() {
         return {
-            custom: (LEVEL_ROLE_MAP[this.props.value] === undefined),
+            levelRoleMap: {},
+            reverseRoles: {},
         };
     },
 
     componentWillMount: function() {
-    	LEVEL_ROLE_MAP = Roles.levelRoleMap();
-    	Object.keys(LEVEL_ROLE_MAP).forEach(function(key) {
-			reverseRoles[LEVEL_ROLE_MAP[key]] = key;
-		});
+        // This needs to be done now because levelRoleMap has translated strings
+        const levelRoleMap = Roles.levelRoleMap();
+        const reverseRoles = {};
+        Object.keys(levelRoleMap).forEach(function(key) {
+            reverseRoles[levelRoleMap[key]] = key;
+        });
+        this.setState({
+            levelRoleMap,
```

Comment:
Hmm - possibly these should just be members rather than state given they don't affect what gets rendered?

Reply:
oh, you mean `reverseRoles`? If it were a member, it wouldn't be defined until the component had mounted. I guess we could make it an ES6, and then define it as a member in the constructor?

Reply:
Well, you could do `this.reverseRoles = ...` on one of the lines above here, although actually I guess technically we do use both these things as part of a render, so ignore me, it probably is better to have them in state.

Reply:
I don't think `reverseRoles` is used as part of a render, only part of `getValue`.

Reply:
Oh you're right, I was assuming getValue was used in a render.